File: Project-source/terrane-admin-web/src/lib/branding.ts
/** Public branding (white-label) API (/admin-api/v1/branding) — no login required, available in the locked state.
 *  Lets the Logo / login page / title bar show the deployer's branding before authentication (product name / theme color / login subtitle). */

import { request } from "@/lib/api";

export interface Branding {
  product_name: string;
  logo_data: string | null;     // Console / workspace Logo
  login_logo: string | null;    // Login page Logo
  favicon: string | null;       // Site favicon
  accent_color: string;
  login_subtitle: string | null;
  support_url: string | null;
  enabled: boolean;
}

export const DEFAULT_BRANDING: Branding = {
  product_name: "Terrane",
  logo_data: null,
  login_logo: null,
  favicon: null,
  accent_color: "#0f9b8e",
  login_subtitle: null,
  support_url: null,
  enabled: true,
};

export function getBranding(): Promise<Branding> {
  return request<Branding>("/admin-api/v1/branding", { credentials: "include" });
}

File: Project-source/terrane-admin-server/app/api/admin_v1/branding.py
"""Public branding (white-label) endpoint (/admin-api/v1/branding) — no login required, available even in the locked state.

The login page / activation page / sidebar must show the deployer's branding (product name / accent color / login subtitle) before authentication,
so this endpoint has no auth and is on the license_gate allowlist (still returns in the locked state). Read-only; falls back to factory defaults when missing
(page-based zero-config rule). Writes go through the PATCH/POST endpoints of settings/wizard (authenticated).
"""

from __future__ import annotations

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.db.platform import get_platform_db
from app.services.platform_settings import get_branding

router = APIRouter(prefix="/admin-api/v1/branding", tags=["branding"])


@router.get("")
async def public_branding(pdb: AsyncSession = Depends(get_platform_db)) -> dict:
    b = await get_branding(pdb)
    return {
        "product_name": b.product_name,
        "logo_data": b.logo_data,
        "login_logo": b.login_logo,
        "favicon": b.favicon,
        "accent_color": b.accent_color,
        "login_subtitle": b.login_subtitle,
        "support_url": b.support_url,
        "enabled": b.enabled,
    }
